Error in catkin_make for Zed wrapper

i am having trouble installing zed wrapper

nvidia@tegra-ubuntu:~/zedwrapper_ws/src$ git clone https://github.com/stereolabs/zed-ros-wrapper.git
nvidia@tegra-ubuntu:~/zedwrapper_ws/src$ cd ..
nvidia@tegra-ubuntu:~/zedwrapper_ws$  #14 

Base path: /home/nvidia/zedwrapper_ws
Source space: /home/nvidia/zedwrapper_ws/src
Build space: /home/nvidia/zedwrapper_ws/build
Devel space: /home/nvidia/zedwrapper_ws/devel
Install space: /home/nvidia/zedwrapper_ws/install
####
#### Running command: "cmake /home/nvidia/zedwrapper_ws/src -DCATKIN_DEVEL_PREFIX=/home/nvidia/zedwrapper_ws/devel -DCMAKE_INSTALL_PREFIX=/home/nvidia/zedwrapper_ws/install -G Unix Makefiles" in "/home/nvidia/zedwrapper_ws/build"
####
-- Using CATKIN_DEVEL_PREFIX: /home/nvidia/zedwrapper_ws/devel
-- Using CMAKE_PREFIX_PATH: /home/nvidia/octomap_ws/devel;/home/nvidia/aerolion_ws/devel;/opt/ros/kinetic
-- This workspace overlays: /home/nvidia/octomap_ws/devel;/home/nvidia/aerolion_ws/devel;/opt/ros/kinetic
-- Using PYTHON_EXECUTABLE: /usr/bin/python
-- Using Debian Python package layout
-- Using empy: /usr/bin/empy
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/nvidia/zedwrapper_ws/build/test_results
-- Found gmock sources under '/usr/src/gmock': gmock will be built
-- Found gtest sources under '/usr/src/gmock': gtests will be built
-- Using Python nosetests: /usr/bin/nosetests-2.7
-- catkin 0.7.11
-- BUILD_SHARED_LIBS is on
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- ~~  traversing 8 packages in topological order:
-- ~~  - zed_depth_sub_tutorial
-- ~~  - zed_display_rviz
-- ~~  - zed_nodelet_example
-- ~~  - zed_ros (metapackage)
-- ~~  - zed_rtabmap_example
-- ~~  - zed_tracking_sub_tutorial
-- ~~  - zed_video_sub_tutorial
-- ~~  - zed_wrapper
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- +++ processing catkin package: 'zed_depth_sub_tutorial'
-- ==> add_subdirectory(zed-ros-wrapper/tutorials/zed_depth_sub_tutorial)
-- +++ processing catkin package: 'zed_display_rviz'
-- ==> add_subdirectory(zed-ros-wrapper/zed_display_rviz)
-- +++ processing catkin package: 'zed_nodelet_example'
-- ==> add_subdirectory(zed-ros-wrapper/examples/zed_nodelet_example)
-- +++ processing catkin metapackage: 'zed_ros'
-- ==> add_subdirectory(zed-ros-wrapper/zed_ros)
-- +++ processing catkin package: 'zed_rtabmap_example'
-- ==> add_subdirectory(zed-ros-wrapper/examples/zed_rtabmap_example)
-- +++ processing catkin package: 'zed_tracking_sub_tutorial'
-- ==> add_subdirectory(zed-ros-wrapper/tutorials/zed_tracking_sub_tutorial)
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
-- +++ processing catkin package: 'zed_video_sub_tutorial'
-- ==> add_subdirectory(zed-ros-wrapper/tutorials/zed_video_sub_tutorial)
-- +++ processing catkin package: 'zed_wrapper'
-- ==> add_subdirectory(zed-ros-wrapper/zed_wrapper)
-- Found CUDA: /usr/local/cuda-9.0 (found suitable exact version "9.0") 
-- Found CUDA: /usr/local/cuda-9.0 (found version "9.0") 
-- Try OpenMP C flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Try OpenMP CXX flag = [-fopenmp]
-- Performing Test OpenMP_FLAG_DETECTED
-- Performing Test OpenMP_FLAG_DETECTED - Success
-- Found OpenMP: -fopenmp  
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
-- zed_wrapper: 0 messages, 3 services
-- Configuring done
-- Generating done
-- Build files have been written to: /home/nvidia/zedwrapper_ws/build
####
#### Running command: "make -j4 -l4" in "/home/nvidia/zedwrapper_ws/build"
####
Scanning dependencies of target zed_wrapper_gencfg
Scanning dependencies of target zed_depth_sub
Scanning dependencies of target zed_video_sub
Scanning dependencies of target zed_tracking_sub
[  3%] Generating dynamic reconfigure files from cfg/Zed.cfg: /home/nvidia/zedwrapper_ws/devel/include/zed_wrapper/ZedConfig.h /home/nvidia/zedwrapper_ws/devel/lib/python2.7/dist-packages/zed_wrapper/cfg/ZedConfig.py
[  6%] Building CXX object zed-ros-wrapper/tutorials/zed_depth_sub_tutorial/CMakeFiles/zed_depth_sub.dir/src/zed_depth_sub_tutorial.cpp.o
[ 10%] Building CXX object zed-ros-wrapper/tutorials/zed_video_sub_tutorial/CMakeFiles/zed_video_sub.dir/src/zed_video_sub_tutorial.cpp.o
[ 13%] Building CXX object zed-ros-wrapper/tutorials/zed_tracking_sub_tutorial/CMakeFiles/zed_tracking_sub.dir/src/zed_tracking_sub_tutorial.cpp.o
Generating reconfiguration files for Zed in zed_wrapper
Wrote header file in /home/nvidia/zedwrapper_ws/devel/include/zed_wrapper/ZedConfig.h
[ 13%] Built target zed_wrapper_gencfg
Scanning dependencies of target _zed_wrapper_generate_messages_check_deps_reset_tracking
[ 13%] Built target _zed_wrapper_generate_messages_check_deps_reset_tracking
Scanning dependencies of target _zed_wrapper_generate_messages_check_deps_set_initial_pose
[ 13%] Built target _zed_wrapper_generate_messages_check_deps_set_initial_pose
Scanning dependencies of target _zed_wrapper_generate_messages_check_deps_reset_odometry
[ 13%] Built target _zed_wrapper_generate_messages_check_deps_reset_odometry
Scanning dependencies of target zed_wrapper_generate_messages_cpp
[ 17%] Generating C++ code from zed_wrapper/reset_odometry.srv
[ 20%] Generating C++ code from zed_wrapper/set_initial_pose.srv
[ 24%] Generating C++ code from zed_wrapper/reset_tracking.srv
[ 24%] Built target zed_wrapper_generate_messages_cpp
Scanning dependencies of target zed_wrapper_generate_messages_lisp
[ 27%] Generating Lisp code from zed_wrapper/reset_odometry.srv
[ 31%] Generating Lisp code from zed_wrapper/set_initial_pose.srv
[ 34%] Generating Lisp code from zed_wrapper/reset_tracking.srv
[ 34%] Built target zed_wrapper_generate_messages_lisp
Scanning dependencies of target zed_wrapper_generate_messages_nodejs
[ 37%] Generating Javascript code from zed_wrapper/reset_odometry.srv
[ 41%] Generating Javascript code from zed_wrapper/set_initial_pose.srv
[ 44%] Generating Javascript code from zed_wrapper/reset_tracking.srv
[ 44%] Built target zed_wrapper_generate_messages_nodejs
Scanning dependencies of target zed_wrapper_generate_messages_eus
[ 48%] Generating EusLisp code from zed_wrapper/reset_odometry.srv
[ 51%] Generating EusLisp code from zed_wrapper/set_initial_pose.srv
[ 55%] Generating EusLisp code from zed_wrapper/reset_tracking.srv
[ 58%] Generating EusLisp manifest code for zed_wrapper
[ 58%] Built target zed_wrapper_generate_messages_eus
Scanning dependencies of target zed_wrapper_generate_messages_py
[ 62%] Generating Python code from SRV zed_wrapper/reset_odometry
[ 65%] Generating Python code from SRV zed_wrapper/set_initial_pose
[ 68%] Generating Python code from SRV zed_wrapper/reset_tracking
[ 72%] Generating Python srv __init__.py for zed_wrapper
[ 75%] Linking CXX executable /home/nvidia/zedwrapper_ws/devel/lib/zed_video_sub_tutorial/zed_video_sub
[ 75%] Built target zed_wrapper_generate_messages_py
Scanning dependencies of target ZEDWrapper
[ 79%] Linking CXX executable /home/nvidia/zedwrapper_ws/devel/lib/zed_depth_sub_tutorial/zed_depth_sub
[ 82%] Building CXX object zed-ros-wrapper/zed_wrapper/CMakeFiles/ZEDWrapper.dir/src/tools/src/sl_tools.cpp.o
[ 82%] Built target zed_video_sub
[ 86%] Building CXX object zed-ros-wrapper/zed_wrapper/CMakeFiles/ZEDWrapper.dir/src/nodelet/src/zed_wrapper_nodelet.cpp.o
[ 86%] Built target zed_depth_sub
Scanning dependencies of target zed_wrapper_generate_messages
[ 86%] Built target zed_wrapper_generate_messages
[ 89%] Linking CXX executable /home/nvidia/zedwrapper_ws/devel/lib/zed_tracking_sub_tutorial/zed_tracking_sub
[ 89%] Built target zed_tracking_sub
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp: In member function ‘virtual void zed_wrapper::ZEDWrapperNodelet::onInit()’:
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:482:41: error: ‘PoseWithCovarianceStamped’ is not a member of ‘geometry_msgs’
             mPubPoseCov = mNh.advertise<geometry_msgs::PoseWithCovarianceStamped>(pose_cov_topic, 1);
                                         ^
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:482:41: error: ‘PoseWithCovarianceStamped’ is not a member of ‘geometry_msgs’
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:482:100: error: no matching function for call to ‘ros::NodeHandle::advertise(std::__cxx11::string&, int)’
             mPubPoseCov = mNh.advertise<geometry_msgs::PoseWithCovarianceStamped>(pose_cov_topic, 1);
                                                                                                    ^
In file included from /opt/ros/kinetic/include/ros/ros.h:45:0,
                 from /home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/include/zed_wrapper_nodelet.hpp:32,
                 from /home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:21:
/opt/ros/kinetic/include/ros/node_handle.h:249:15: note: candidate: template<class M> ros::Publisher ros::NodeHandle::advertise(const string&, uint32_t, bool)
     Publisher advertise(const std::string& topic, uint32_t queue_size, bool latch = false)
               ^
/opt/ros/kinetic/include/ros/node_handle.h:249:15: note:   template argument deduction/substitution failed:
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:482:100: error: template argument 1 is invalid
             mPubPoseCov = mNh.advertise<geometry_msgs::PoseWithCovarianceStamped>(pose_cov_topic, 1);
                                                                                                    ^
In file included from /opt/ros/kinetic/include/ros/ros.h:45:0,
                 from /home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/include/zed_wrapper_nodelet.hpp:32,
                 from /home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:21:
/opt/ros/kinetic/include/ros/node_handle.h:315:13: note: candidate: template<class M> ros::Publisher ros::NodeHandle::advertise(const string&, uint32_t, const SubscriberStatusCallback&, const SubscriberStatusCallback&, const VoidConstPtr&, bool)
   Publisher advertise(const std::string& topic, uint32_t queue_size,
             ^
/opt/ros/kinetic/include/ros/node_handle.h:315:13: note:   template argument deduction/substitution failed:
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:482:100: error: template argument 1 is invalid
             mPubPoseCov = mNh.advertise<geometry_msgs::PoseWithCovarianceStamped>(pose_cov_topic, 1);
                                                                                                    ^
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp: In member function ‘void zed_wrapper::ZEDWrapperNodelet::start_tracking()’:
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:790:25: error: ‘struct sl::TrackingParameters’ has no member named ‘set_floor_as_origin’
             trackParams.set_floor_as_origin = mFloorAlignment;
                         ^
In file included from /opt/ros/kinetic/include/ros/ros.h:40:0,
                 from /home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/include/zed_wrapper_nodelet.hpp:32,
                 from /home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:21:
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:791:69: error: ‘struct sl::TrackingParameters’ has no member named ‘set_floor_as_origin’
             NODELET_INFO_STREAM("Floor Alignment : " << trackParams.set_floor_as_origin);
                                                                     ^
/opt/ros/kinetic/include/ros/console.h:356:64: note: in definition of macro ‘ROSCONSOLE_PRINT_STREAM_AT_LOCATION_WITH_FILTER’
     __rosconsole_print_stream_at_location_with_filter__ss__ << args; \
                                                                ^
/opt/ros/kinetic/include/ros/console.h:398:7: note: in expansion of macro ‘ROSCONSOLE_PRINT_STREAM_AT_LOCATION’
       ROSCONSOLE_PRINT_STREAM_AT_LOCATION(args); \
       ^
/opt/ros/kinetic/include/ros/console.h:568:43: note: in expansion of macro ‘ROS_LOG_STREAM_COND’
 #define ROS_LOG_STREAM(level, name, args) ROS_LOG_STREAM_COND(true, level, name, args)
                                           ^
/opt/ros/kinetic/include/rosconsole/macros_generated.h:113:43: note: in expansion of macro ‘ROS_LOG_STREAM’
 #define ROS_INFO_STREAM_NAMED(name, args) ROS_LOG_STREAM(::ros::console::levels::Info, std::string(ROSCONSOLE_NAME_PREFIX) + "." + name, args)
                                           ^
/opt/ros/kinetic/include/nodelet/nodelet.h:60:34: note: in expansion of macro ‘ROS_INFO_STREAM_NAMED’
 #define NODELET_INFO_STREAM(...) ROS_INFO_STREAM_NAMED(getName(), __VA_ARGS__)
                                  ^
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:791:13: note: in expansion of macro ‘NODELET_INFO_STREAM’
             NODELET_INFO_STREAM("Floor Alignment : " << trackParams.set_floor_as_origin);
             ^
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp: In member function ‘void zed_wrapper::ZEDWrapperNodelet::publishPose(ros::Time)’:
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:896:17: error: ‘PoseWithCovarianceStamped’ is not a member of ‘geometry_msgs’
                 geometry_msgs::PoseWithCovarianceStamped poseCov;
                 ^
/home/nvidia/zedwrapper_ws/src/zed-ros-wrapper/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp:898:17: error: ‘poseCov’ was not declared in this scope
                 poseCov.header = header;
                 ^
zed-ros-wrapper/zed_wrapper/CMakeFiles/ZEDWrapper.dir/build.make:86: recipe for target 'zed-ros-wrapper/zed_wrapper/CMakeFiles/ZEDWrapper.dir/src/nodelet/src/zed_wrapper_nodelet.cpp.o' failed
make[2]: *** [zed-ros-wrapper/zed_wrapper/CMakeFiles/ZEDWrapper.dir/src/nodelet/src/zed_wrapper_nodelet.cpp.o] Error 1
CMakeFiles/Makefile2:2064: recipe for target 'zed-ros-wrapper/zed_wrapper/CMakeFiles/ZEDWrapper.dir/all' failed
make[1]: *** [zed-ros-wrapper/zed_wrapper/CMakeFiles/ZEDWrapper.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
Invoking "make -j4 -l4" failed

--- a/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp
+++ b/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp
@@ -751,13 +751,13 @@ namespace zed_wrapper {
     }
 
     void ZEDWrapperNodelet::start_tracking() {
-        NODELET_INFO_STREAM("Starting Tracking");
+        NODELET_INFO_STREAM("*** Starting Positional Tracking ***");
         mNhNs.getParam("odometry_DB", mOdometryDb);
         mNhNs.getParam("pose_smoothing", mPoseSmoothing);
         mNhNs.getParam("spatial_memory", mSpatialMemory);
         mNhNs.getParam("floor_alignment", mFloorAlignment);
         mNhNs.getParam("init_odom_with_first_valid_pose", mInitOdomWithPose);
-        NODELET_INFO_STREAM("Init Odometry with first valid pose data : " << mInitOdomWithPose);
+        NODELET_INFO_STREAM("Init Odometry with first valid pose data : " << (mInitOdomWithPose ? "ENABLED" : "DISABLED"));
 
         if (mInitialTrackPose.size() != 6) {
             NODELET_WARN_STREAM("Invalid Initial Pose size (" << mInitialTrackPose.size()
@@ -779,9 +779,9 @@ namespace zed_wrapper {
         sl::TrackingParameters trackParams;
         trackParams.area_file_path = mOdometryDb.c_str();
         trackParams.enable_pose_smoothing = mPoseSmoothing;
-        NODELET_INFO_STREAM("Pose Smoothing : " << trackParams.enable_pose_smoothing);
+        NODELET_INFO_STREAM("Pose Smoothing : " << (trackParams.enable_pose_smoothing ? "ENABLED" : "DISABLED"));
         trackParams.enable_spatial_memory = mSpatialMemory;
-        NODELET_INFO_STREAM("Spatial Memory : " << trackParams.enable_spatial_memory);
+        NODELET_INFO_STREAM("Spatial Memory : " << (trackParams.enable_spatial_memory ? "ENABLED" : "DISABLED"));
         trackParams.initial_world_transform = mInitialPoseSl;
 
         if (mFloorAlignment &&
@@ -789,7 +789,7 @@ namespace zed_wrapper {
             NODELET_WARN("Floor Alignment is available starting from SDK v2.6");
         } else {
             trackParams.set_floor_as_origin = mFloorAlignment;
-            NODELET_INFO_STREAM("Floor Alignment : " << trackParams.set_floor_as_origin ? "ENABLED" : "DISABLED");
+            NODELET_INFO_STREAM("Floor Alignment : " << (trackParams.set_floor_as_origin ? "ENABLED" : "DISABLED"));
         }
 
         mZed.enableTracking(trackParams);
@@ -1802,7 +1802,10 @@ namespace zed_wrapper {
                         } else {
                             NODELET_DEBUG_STREAM("ODOM -> Tracking Status: " << sl::toString(status));
                         }
+                    } else if (mFloorAlignment) {
+                        NODELET_WARN_THROTTLE(5.0, "Odometry will be published as soon as the floor as been detected for the first time");
                     }
+
                 }
 
                 // Publish the zed camera pose if someone has subscribed to

--- a/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp
+++ b/zed_wrapper/src/nodelet/src/zed_wrapper_nodelet.cpp
@@ -32,6 +32,7 @@
 #include <sensor_msgs/image_encodings.h>
 #include <sensor_msgs/point_cloud2_iterator.h>
 #include <stereo_msgs/DisparityImage.h>
+#include <geometry_msgs/PoseWithCovarianceStamped.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
 
 using namespace std;
@@ -788,7 +789,7 @@ namespace zed_wrapper {
             NODELET_WARN("Floor Alignment is available starting from SDK v2.6");
         } else {
             trackParams.set_floor_as_origin = mFloorAlignment;
-            NODELET_INFO_STREAM("Floor Alignment : " << trackParams.set_floor_as_origin);
+            NODELET_INFO_STREAM("Floor Alignment : " << trackParams.set_floor_as_origin ? "ENABLED" : "DISABLED");
         }
 
         mZed.enableTracking(trackParams);

